libdevcore/MiniMoustache.h

```diff
@@ -0,0 +1,92 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file MiniMoustache.h
+ * @author Chris <chis@ethereum.org>
+ * @date 2017
+ *
+ * Moustache-like templates.
+ */
+
+#pragma once
+
+#include <libdevcore/Exceptions.h>
+
+#include <string>
+#include <map>
+#include <vector>
+
+namespace dev
+{
+
+DEV_SIMPLE_EXCEPTION(MiniMoustacheError);
+
+///
+/// Moustache-like templates.
+///
+/// Usage:
+///     std::vector<std::map<std::string, std::string>> listValues(2);
+///     listValues[0]["k"] = "key1";
+///     listValues[0]["v"] = "value1";
+///     listValues[1]["k"] = "key2";
+///     listValues[1]["v"] = "value2";
+///     auto s = MiniMoustache("<p1>\n<#list><k> -> <v>\n</list>")
```

Comment:
Gotcha. I guess we should rename it to reflect that it has a different syntax, *mini* to me signals it is a partial implementation, but the syntax has a major differnece :)

Some ideas *Moustachio*, Moustaki, Whiskers.

, perhaps *Mustachio* ?